stdpopsim/slim_engine.py

```diff
@@ -602,6 +603,63 @@
 
 """
 
+_raw_stdpopsim_top_level_schema = {
+    "stdpopsim": {
+        "description": "Top-level metadata for a tree sequence produced by stdposim using the SLiM engine",  # noqa: E501
+        "type": "object",
+        "properties": {
+            "DFEs": {
+                "type": "array",
+                "items": {
+                    "type": "object",
+                    "properties": {
+                        "id": {"type": "string"},
+                        "description": {"type": "string"},
+                        "long_description": {"type": "string"},
+                        "mutation_types": {
+                            "type": "array",
+                            "items": {
+                                "type": "object",
+                                "properties": {
+                                    "dominance_coeff": {"type": "number"},
+                                    "distribution_type": {"type": "string"},
+                                    "distribution_args": {
+                                        "type": "array",
+                                        "items": {"type": ["number", "string"]},
+                                    },
+                                    "convert_to_substitution": {"type": "boolean"},
+                                    "Q_scaled_index": {
+                                        "type": "array",
+                                        "items": {"type": "number"},
+                                    },
+                                    "slim_mutation_type_id": {"type": "integer"},
+                                    "is_neutral": {"type": "boolean"},
+                                    "simulated_in_SLiM": {"type": "boolean"},
+                                },
+                            },
+                        },
+                        "proportions": {"type": "array", "items": {"type": "number"}},
+                        "slim_proportions": {
+                            "type": "array",
+                            "items": {"type": "number"},
+                        },
```

Comment:
```suggestion
```
This seems like an internal detail that shouldn't be in metadata?

Reply:
I also thought this at first, but this makes our lives so much easier in the tests (see line 992 of test_slim_engine.py). But I could replicate the behavior independently and get the slim proportions there...

Reply:
If [that's](https://github.com/popsim-consortium/stdpopsim/pull/1152/files#diff-b201385bdd9318912fe29722d750c1d4007aa5fa1454b0d8328eb6cba68dcdadR992) the place we need it then I think replicating is the way to go - currently that bit is only testing our ability to store stuff in metadata, not that we've got the proportions right.

Reply:
yeah, agreed! now I replicated the behavior.